CRM/Sqltasks/BAO/SqltasksActionTemplate.php

```diff
@@ -0,0 +1,119 @@
+<?php
+use CRM_Sqltasks_ExtensionUtil as E;
+
+class CRM_Sqltasks_BAO_SqltasksActionTemplate extends CRM_Sqltasks_DAO_SqltasksActionTemplate {
```

Comment:
Most of the methods in this class are used in the corresponding API3 code. It's generally not necessary to write this low-level code as `civix generate:entity` with `--api-version 3` (or `--api-version 3,4`) will auto-generated API functions/classes for all CRUD operations. The generated code for API3 may look similar to [this](https://github.com/greenpeace-cee/dialfire/blob/main/api/v3/DialfireCampaign.php).

We can probably remove most of this code and the custom API implementation files (i.e. `api/v3/SqltasksActionTemplate/Create.php` and others) if you re-run `civix generate:entity SqltasksActionTemplate --api-version 3`. The [documentation](https://docs.civicrm.org/dev/en/latest/extensions/civix/#generate-entity) describes this process quite well.

Note: Please see my comments re: the schema files below before doing that.

Reply:
Resolved with commit 1d6fc5bd5f0ea6a622db81cccb2fac28b02da05c [Updated api calls and result values sequential]